pkg/kubectl/cmd/config/create_authinfo.go

```diff
@@ -129,9 +130,9 @@ func (o *createAuthInfoOptions) modifyAuthInfo(existingAuthInfo clientcmdapi.Aut
 
 	var setToken, setBasic bool
 
-	if o.clientCertificate.Provided() {
```

Comment:
you need a way to distinguish between unspecified and specified-and-empty... string pointer, Provided(), etc